unittest flake: TestControllerSync Warning ClaimLost Bound claim has lost its PV

https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/26335/kubernetes-pull-test-unit-integration/28566/artifacts/
```
--- FAIL: TestControllerSync (0.13s)
	framework_test.go:395: Unexpected event received: "Warning ClaimLost Bound claim has lost its PersistentVolume. Data on the volume is lost!"
	framework_test.go:739: Test "5-4 - delete volume": Events do not match
```
@kubernetes/sig-storage @jsafrane please close if it's fixed by https://github.com/kubernetes/kubernetes/pull/26555

--- a/pkg/controller/persistentvolume/framework_test.go
+++ b/pkg/controller/persistentvolume/framework_test.go
@@ -358,11 +358,14 @@ func (r *volumeReactor) checkClaims(t *testing.T, expectedClaims []*api.Persiste
 func checkEvents(t *testing.T, expectedEvents []string, ctrl *PersistentVolumeController) error {
 	var err error
 
-	// Read recorded events
+	// Read recorded events - wait up to 1 minute to get all the expected ones
+	// (just in case some goroutines are slower with writing)
+	timer := time.NewTimer(time.Minute)
+
 	fakeRecorder := ctrl.eventRecorder.(*record.FakeRecorder)
 	gotEvents := []string{}
 	finished := false
-	for !finished {
+	for len(gotEvents) < len(expectedEvents) && !finished {
 		select {
 		case event, ok := <-fakeRecorder.Events:
 			if ok {
@@ -372,8 +375,8 @@ func checkEvents(t *testing.T, expectedEvents []string, ctrl *PersistentVolumeCo
 				glog.V(5).Infof("event recorder finished")
 				finished = true
 			}
-		default:
-			glog.V(5).Infof("event recorder finished")
+		case _, _ = <-timer.C:
+			glog.V(5).Infof("event recorder timeout")
 			finished = true
 		}
 	}

